### PR TITLE
feat!: require Node 22, test on Node 26, align CI with .nvmrc

### PR DESCRIPTION
**Pull request by Texarkanine**
# Pull Request

## Summary

1. **Breaking change:** Raise minimum Node.js to 22 (`engines`, optional `@types/node` peer, dev `@types/node`) in [package.json](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/package.json); refresh [package-lock.json](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/package-lock.json).
2. **CI:** PR matrix tests Node [22, 24, 26](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/.github/workflows/pr.yaml); Codecov and `attw` run on the lowest matrix job (22).
3. **Single source for workflow Node:** Add [.nvmrc](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/.nvmrc); [release-please.yaml](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/.github/workflows/release-please.yaml) and [generate-demos.yaml](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/.github/workflows/generate-demos.yaml) use `node-version-file: .nvmrc`.
4. **Demos:** [demos/Dockerfile](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/demos/Dockerfile) installs Node 22 via NodeSource `setup_22.x`.
5. **Docs:** [README.md](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/README.md), [CONTRIBUTING.md](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/CONTRIBUTING.md), and memory-bank [productContext.md](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/memory-bank/productContext.md) / [techContext.md](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/memory-bank/techContext.md) updated for Node 22+.
6. **Vitest:** Remove unused `esbuild` block from [vitest.config.ts](https://github.com/Texarkanine/inquirerjs-checkbox-search/blob/node-26/vitest.config.ts) (Vitest 4 uses oxc; the block only produced a warning).

## Details

Implements [issue #129](https://github.com/Texarkanine/inquirerjs-checkbox-search/issues/129): drop EOL Node 20, minimum Node 22, add Node 26 to the PR matrix, and align release/demo CI with `.nvmrc` so future baseline bumps are mostly a one-line `.nvmrc` change for those workflows.

## Testing

Ran `npm test` (format, lint, typecheck, full Vitest suite) on Node 22; all 113 tests passed.

### Test Cases Covered

- [ ] Unit tests added/updated for new functionality
- [ ] Edge cases tested
- [ ] Error conditions handled
- [ ] User interaction flows verified

No production code changed; existing unit tests cover library behavior. Full suite green on Node 22.

### Manual Testing

None required beyond the automated suite.

## Screenshots/Recordings

N/A (no UI changes).

## Checklist

- [x] Commit message follows [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Commit message clearly describes the change
- [x] Includes breaking change notation if applicable

## Summary by CodeRabbit

* **Chores**
  * Updated minimum Node.js version requirement to 22+ for enhanced compatibility
  * Expanded CI testing infrastructure to validate against Node.js versions 22, 24, and 26
  * Updated development documentation and configuration to reflect new Node.js version requirements

[Review Change Stack](https://app.coderabbit.ai/change-stack/Texarkanine/inquirerjs-checkbox-search/pull/131)

### CODE REVIEW

File: demos/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN sed -i 's/stable/bookworm/g' /etc/apt/sources.list.d/*
 
 RUN apt-get update && apt-get install -y \
     curl \
-    && curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
+    && curl -fsSL https://deb.nodesource.com/setup_22.x | bash - \
```
Comment (Texarkanine) [inline demos/Dockerfile:11]: Note that the previews are now generated with Node 22 instead of Node 20.
